homework3/semi-uber/Server/api/handlers/user.handlers/userPatchHandler.js

```diff
@@ -0,0 +1,34 @@
+const config = require("config");
+const log4js = require("log4js");
+const logger = log4js.getLogger();
+const { User } = require("../../Schemas/user.schema");
+const bcrypt = require("bcrypt");
+const saltRounds = Number(config.get("saltRounds"));
+const errorHandler = require("../error.handler");
+
+const hashPassword = (password, saltRounds) => {
+  return bcrypt.hash(password, saltRounds);
+};
+
+async function userPatchHandler(req, res) {
+  const { newPassword, id: userID } = req.body;
+  const hashedPass = await hashPassword(newPassword, saltRounds);
+  if (!hashPassword) {
+    return errorHandler("Password chanching was failt!", res);
+  }
+  const updatedUser = await User.findByIdAndUpdate(userID, {
```

Comment:
Same here, I guess you need try/catch for such code. `if (!updatedUser)` will not catch an error, after something goes wrong on this line, the code execution will no go further.

Reply:
Fixed